samtranslator/metrics/method_decorator.py

```diff
@@ -0,0 +1,110 @@
+"""
+Method decorator for execution latency collection
+"""
+import functools
+from datetime import datetime
+from samtranslator.metrics.metrics import DummyMetricsPublisher, Metrics
+from samtranslator.model import Resource
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class MetricsMethodWrapperSingleton:
+    """
+    Keeps the instance of Metrics object.
+    This singleton will be alive until lambda receives shutdown event
+    """
+
+    _METRICS_INSTANCE = None
+    _DUMMY_INSTANCE = Metrics("ServerlessTransform", DummyMetricsPublisher())
+
+    @staticmethod
+    def set_instance(metrics):
+        MetricsMethodWrapperSingleton._METRICS_INSTANCE = metrics
+
+    @staticmethod
+    def get_instance():
+        """
+        Return the instance, if nothing is set return a dummy one
+        """
+        if MetricsMethodWrapperSingleton._METRICS_INSTANCE:
+            return MetricsMethodWrapperSingleton._METRICS_INSTANCE
+
+        return MetricsMethodWrapperSingleton._DUMMY_INSTANCE
+
+
+def _get_metric_name(prefix, name, func, args):
+    """
+    Returns the metric name depending on the parameters
+
+    Parameters
+    ----------
+    prefix : str
+        A string that will always be added in the beginning of metric name.
+    name : str
+        The name of the metric. If None is given, it will try to read from function and argument details.
+    func : Function
+        The function that is decorated. This will be used as metric name if name is not provided and caller is not an
+        instance of Resource object.
+    args : args
+        Arguments that is originally passed to the caller. This function will check if first element in this function
+        is a Resource then it reads the 'resource_type' property out of it to generate the metric name.
+    """
+    if name:
+        metric_name = name
+    else:
+        if args > 0 and isinstance(args[0], Resource):
```

Comment:
Should this be `len(args) > 0`?

Reply:
Shoot, I was converting this from `if len(args) > 0` to `if args`. Let me fix this.